ratis-shell/src/main/java/org/apache/ratis/shell/cli/sh/command/GroupCommand.java

```diff
@@ -17,38 +17,39 @@
  */
 package org.apache.ratis.shell.cli.sh.command;
 
-import org.apache.commons.cli.CommandLine;
 import org.apache.commons.cli.Option;
 import org.apache.commons.cli.Options;
-import org.apache.ratis.client.RaftClient;
-import org.apache.ratis.protocol.RaftClientReply;
-import org.apache.ratis.protocol.RaftPeer;
-import org.apache.ratis.protocol.RaftPeerId;
-import org.apache.ratis.shell.cli.RaftUtils;
+import org.apache.ratis.shell.cli.Command;
+import org.apache.ratis.shell.cli.sh.group.GroupInfoCommand;
+import org.apache.ratis.shell.cli.sh.group.GroupListCommand;
 
-import java.io.IOException;
-import java.net.InetSocketAddress;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.function.BiConsumer;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import java.util.function.Function;
 
 /**
- * Command for remove and add ratis server.
+ * Command for the ratis group
  */
 public class GroupCommand extends AbstractRatisCommand {
-  public static final String REMOVE_OPTION_NAME = "remove";
-  public static final String ADD_OPTION_NAME = "add";
+
+  private static final Map<String, Function<Context, ? extends Command>>
+          SUB_COMMANDS = new HashMap<>();
+
+  static {
+    SUB_COMMANDS.put("info", GroupInfoCommand::new);
+    SUB_COMMANDS.put("list", GroupListCommand::new);
+  }
+
+  private Map<String, Command> mSubCommands = new HashMap<>();
 
   /**
    * @param context command context
    */
   public GroupCommand(Context context) {
     super(context);
+    SUB_COMMANDS.forEach((name, constructor) -> {
+      mSubCommands.put(name, constructor.apply(context));
+    });
   }
```

Comment:
This can be simplified as below:
```
  private static final List<Function<Context, AbstractRatisCommand>> SUB_COMMAND_CONSTRUCTORS
      = Collections.unmodifiableList(Arrays.asList(
          GroupInfoCommand::new, GroupListCommand::new));

  private final Map<String, Command> subs;

  /**
   * @param context command context
   */
  public GroupCommand(Context context) {
    super(context);
    this.subs = Collections.unmodifiableMap(SUB_COMMAND_CONSTRUCTORS.stream()
        .map(constructor -> constructor.apply(context))
        .collect(Collectors.toMap(Command::getCommandName, Function.identity())));
  }
```